pkg/kritis/cryptolib/README.md

```diff
@@ -0,0 +1,41 @@
+# Crypto Library
+
+The Crypto Library provides an interface for creating and verifying Attestations through a Signer and Verifier, respectively.
+
+## Components
+
+### Attestation
+
+An Attestation is a statement that a container image meets a policy and can be deployed to a Kubernetes cluster. It contains a signature generated by the private key of a trusted entity (e.g. a build system, vulnerability scanner, etc.). However, as anybody can create an Attestation, it cannot be trusted until it is verified using the entity's corresponding public key. An Attestation is defined independently from the wire format of a signature, allowing the user to store signatures generated by PGP or PKIX keys, or signatures encoded as JWTs.
+
+An Attestation contains a payload and a signature generated over the payload with the trusted entity’s private key. It also contains the ID of the public key which can verify the Attestation’s signature.
+
+### Public Key
+A PublicKey is the definitive trust anchor used to verify that an Attestation’s signature is valid. Unlike Attestations, which are considered untrustworthy until verified, PublicKeys are assumed to contain trustworthy information. Consequently, this information should be provided directly by the trusted party.
+
+A PublicKey contains the raw public key material and an ID. It also contains a KeyType, one of {`Pgp`, `Pkix`, or `Jwt`}, indicating how the trusted entity stores data within the Attestation. It also contains a SignatureAlgorithm, indicating the cryptographic algorithm, padding algorithm, and hash function used on the payload to create the signature in the Attestation.
+
+### Private Key
+The trusted entity has a private key, which is used by the Signer to generate an Attestation’s signature.
+
+### Payload
+The payload is a message provided by the trusted entity. It is signed by their private key to create a signature. By convention, the payload is a JSON-encoded string conforming to the [Red Hat Atomic Host signature format](https://github.com/aweiteka/image/blob/e5a20d98fe698732df2b142846d007b45873627f/docs/signature.md).
+
+## Interface
+
+### Signing
+
+#### Signer
+A trusted entity will use a Signer to generate Attestations. There are three signer implementations, one for each KeyType: a `PgpSigner`, `PkixSigner`, and a `JwtSigner`. Each signer has its own constructor (e.g. `NewPgpSigner`), which creates a Signer storing the trusted entity’s private key and any other data required to create an Attestation.
+
+Each signer implements the Signer interface: the `CreateAttestation` method. When passed a payload to sign over, `CreateAttestation` will generate and return an Attestation containing the signature.
+
+### Verifying
+
+#### PublicKey
+To create a PublicKey, the user can call `NewPublicKey`, passing in the public key material, key ID, KeyType, and any other data necessary to verify an Attestation.
```

Comment:
Replacing the text with code will be clearer?

Reply:
We plan on putting code snippets in the tutorial doc (see #569)